packages/jsdoc-plugins/lib/relation-fixer/addmissingdoclets.js

```diff
@@ -2,29 +2,33 @@
  * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
  * Licensed under the terms of the MIT License (see LICENSE.md).
  */
+// @ts-check
 
 'use strict';
 
+const { cloneDeep } = require( 'lodash' );
 const getMissingDocletsData = require( './getmissingdocletsdata' );
-const cloneDeep = require( 'lodash' ).cloneDeep;
 const DocletCollection = require( '../utils/doclet-collection' );
 
 module.exports = addMissingDoclets;
 
 /**
- * Adds missing doclets of members coming from implemented interfaces and extended classes.
- * JSDoc does not support inheritance of static members which is why the plugin was made.
- * This module requires the input to be processed by 'buildrelations' module first.
+ * Adds missing doclets for members coming from implemented interfaces, extended classes, and mixins.
+ * It does also support inheriting static members and typedef inheritance, which both are not supported by the JSDoc.
+ * This module requires the input preprocessed by the `buildRelations()` function.
  *
- * @param {Array.<Doclet>} originalDoclets
+ * @param {Array.<Doclet>} doclets
  * @returns {Array.<Doclet>}
  */
-function addMissingDoclets( originalDoclets ) {
-	const clonedDoclets = cloneDeep( originalDoclets );
+function addMissingDoclets( doclets ) {
+	doclets = cloneDeep( doclets );
+
 	const docletCollection = new DocletCollection();
+
+	/** @type {Doclet[]} */
 	const typedefDoclets = [];
 
-	for ( const doclet of clonedDoclets ) {
+	for ( const doclet of doclets ) {
```

Comment:
Would be good to avoid overwriting the function's parameters. I mean the change in L24 for me is invalid.

If you would like to simplify the thing, you could write something like this:

`for ( const doclet of cloneDeep( doclets ) ) {`

Reply:
IMO overriding parameters is totally fine if you want to be sure that you won't make changes to the original parameters.

Reply:
I don't agree. Overriding parameters always leads to unexpected results. It may not happen in this situation but we should avoid such danger.

Reply:
why?

Reply:
what bad can happen?